core/src/main/java/org/apache/accumulo/core/util/Nulls.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.util;
+
+import java.util.function.Function;
+
+public class Nulls {
+  public static <T,R> R map(T object, Function<T,R> conversion) {
+    return object == null ? null : conversion.apply(object);
+  }
```

Comment:
I'm not sure we really need a class for this, but as long as it's here, I think a more clear name might be helpful:

```suggestion
public class Nulls {
  public static <T,R> R mapIfNotNull(T object, Function<T,R> conversion) {
    return object == null ? null : conversion.apply(object);
  }
```

Reply:
I created it because using optional is long and creates an object. Using the java ternary operator is also long and requires repeating the variable which is error prone. Below are a comparison of the alternatives.

```java
Text firstSplit = Nulls.map(info.firstSplit, Text::new);
```

```java
Text firstSplit = info.firstSplit == null ? null : new Text(info.firstSplit);
```

```java
Text firstSplit = Optional.ofNullable(info.firstSplit).map(Text::new).orElse(null);
```

As for the name, given the class name is Nulls I did not think the method name needed to be more descriptive.

Reply:
I wouldn't use the Optional approach, but I'm pretty used to seeing the second option, and find it to be an acceptable level of boilerplate and risk. But, I'm also fine with the method. I still think it warrants a better name. I don't find the class name particularly helpful for explaining what the method does. If anything, it seems to imply that the method is mapping null values, not non-null ones, which is the exact opposite of what it's doing. Maybe if it was `NotNull.map()` or `NonNull.map()`, it would align to how it's being used here. But, I think the better method name would suffice, so that the `Nulls` class could be used for other convenient static methods as well.

It may also be useful to add a private constructor on `Null` to avoid anybody instantiating it, since it's just a utility class for static methods.

Reply:
I think I will remove the Nulls class as its tangential to the purpose of this PR.  Its something I like, I but I only like it because its short and the longer its name is the less I like it.

Reply:
removed in 3edd638

Reply:
> ...I only like it because its short...

A static import could keep the line short. I also thought of `nullOrMap`, which has the same line length as your original, with the static import. The length of the class name doesn't even matter, then, and you could put the method in any utility class.

I'm not always a fan of static imports, but I think there's a few cases where they really help with boilerplate utility code. This seems like it could be one of those cases. We do use the pattern in a lot of places, and having an error-resistant implementation for reuse somewhere might be helpful. Could be done as a separate PR for all of Accumulo.